### PR TITLE
feat: [UX-332] Add multiselect support

### PR DESCRIPTION
**Pull request by witwash**
#### Summary

- Added `multiselect` support for the `OptimusSelectInput`
- Refactored `OptimusSelectInput`
- Updated story

<details><summary>Video</summary>

https://github.com/MewsSystems/mews-flutter/assets/9210422/3ba2732e-a7d2-4a0e-84e1-4f845573842f

</details>

#### Testing steps

- Open` Select Input` story
- Enable `multiselect`
- Test the component using different variants of exposed options.

#### Follow-up issues

Refactoring 🔧 

#### Check during review

- Verify against Jira issue.
- Is the PR over 300 additions? Consider rejecting it with advice to split it. Is it over 500 additions? It should definitely be rejected.
- Unused code removed.
- Build passing.
- Is it a bug fix? Check that it is covered by a proper test (unit or integration).

### CODE REVIEW

**Comment by ookami-kb** [comment]
@witwash can refactoring be done as a separate PR to make this PR smaller?

**Comment by witwash** [comment]
@ookami-kb, here is just the `multiselect` part. The refactoring will come in the follow-up. This is kinda the best I can do. New `multiselect` is using the same parts as the regular input, but is not input per se, so I had to take out some common parts.

**Comment by ookami-kb** [comment]
@witwash ah, I thought you could do refactoring PR with extracting some "future"-common part as a pre-step.

**Comment by witwash** [comment]
@ookami-kb, can you look at this one please? 🙏

File: optimus/lib/src/select_input.dart [hunk 1]
```diff
@@ -90,6 +92,14 @@ class OptimusSelectInput<T> extends StatefulWidget {
   /// {@endtemplate}
   final GroupBuilder? groupBuilder;
 
+  /// Whether to enable having several items selected at the same time.
```
Comment (ookami-kb) [inline optimus/lib/src/select_input.dart:95]: I would rephrase it to something like

> If `true`, you can select multiple items at the same time.

File: optimus/lib/src/select_input.dart [hunk 2]
```diff
@@ -90,6 +92,14 @@ class OptimusSelectInput<T> extends StatefulWidget {
...
+  /// State of the selected items is manages outside this widget and has to be
```
Comment (ookami-kb) [inline optimus/lib/src/select_input.dart:96]:
```suggestion
  /// State of the selected items is managed outside this widget and has to be
```

File: optimus/lib/src/form/multiselect_field.dart [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+  void dispose() {
+    _effectiveFocusNode.removeListener(_onStateUpdate);
+    _focusNode?.dispose();
+    super.dispose();
+  }
+
+  bool get _isUsingInlineError =>
+      widget.errorVariant == OptimusInputErrorVariant.inlineTooltip;
+
+  void _onStateUpdate() => setState(() {});
```
Comment (ookami-kb) [inline optimus/lib/src/form/multiselect_field.dart:82]: `_handleStateUpdate`

**Review by ookami-kb — APPROVED** [review]
Minor comments, otherwise, LGTM.
